xonsh/pretty.py

```diff
@@ -0,0 +1,847 @@
+# -*- coding: utf-8 -*-
+"""
+Python advanced pretty printer.  This pretty printer is intended to
+replace the old `pprint` python module which does not allow developers
+to provide their own pretty print callbacks.
+
+This module is based on ruby's `prettyprint.rb` library by `Tanaka Akira`.
+
```

Comment:
Add the IPython copyright notice here on this line, please.  See https://github.com/scopatz/xonsh/blob/master/xonsh/tools.py#L3 for an example.